### PR TITLE
chore: port the remaining doctoral course

### LINKED ISSUE #238: Bring descriptions from doctoral school material
[Doctoral school material](https://github.com/UCL-RITS/doctoral-programming-intro) shares the two sections about Python, but they've diverged a bit over the years. Doctoral school's getting more details in the descriptions. We should bring all that's useful. 

Ideally keeping the git-history, but that will be very difficult, so at least [adding the contributors as co-authors](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors) would be good.

### PR DESCRIPTION
**Pull request by Saransh-cpp**
Closes #238 

Related to "Research Data in Python"

### CODE REVIEW

**Review by dpshelio — CHANGES_REQUESTED** [review]
Perfect! Some small things to update.

File: ch02data/082NumPy.ipynb.py [hunk 1]
```diff
@@ -12,382 +12,471 @@
...
+# MATLAB has typically been the most popular "language of technical computing", with strong built-in support for efficient numerical analysis with matrices (the *mat* in MATLAB is for matrix, not maths), and plotting.
+#
+# Other dynamic languages can be argued to have cleaner, more logical syntax, for example [Ruby](https://www.ruby-lang.org/en/) or [Scheme][scheme].
+#
+# [scheme]: https://en.wikipedia.org/wiki/Scheme_(programming_language)
 # But Python users developed three critical libraries, matching the power of MATLAB for scientific work:
+#
+# * [Matplotlib](https://matplotlib.org/), a plotting library for creating visualizations in Python.
+# * [NumPy](https://numpy.org/), a fast numeric computing library offering a flexible *n*-dimensional array type.
+# * [IPython](https://ipython.readthedocs.io/en/stable/overview.html), an interactive Python interpreter that later led to the [Jupyter notebook](https://jupyter.org/) interface.
```
[the remark this answers is not in the document]
Comment (Saransh-cpp) [reply ch02data/082NumPy.ipynb.py:31]: That works! 🎉

File: ch02data/062basic_plotting.ipynb.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+#
+# The `pyplot` interface also provides functions for adding axis labels and a title to our plot. Specifically `plt.xlabel` and `plt.ylabel` are functions which set the labels on respectively the horizontal (*x*) axis and vertical (*y*) axis, both accepting a string argument corresponding to the axis label. The `plt.title` function, as the name suggests, allows setting an overall title for the plot. As for the legend labels, the axis labels and title may all optionally use TeX mathematical notation delimited by dollar `$` signs.
+
+# %%
+plt.plot(theta, [sin(t) for t in theta], label=r"$f(\theta) = \sin\theta$")
+plt.plot(theta, [cos(t) for t in theta], linestyle="dotted", label=r"$f(\theta) = \cos\theta$")
+plt.legend()
+plt.xlabel(r"Angle in radians $\theta$")
+plt.ylabel(r"$f(\theta)$")
+plt.title("Trigonometric functions")
```
[the remark this answers is not in the document]
Comment (Saransh-cpp) [reply ch02data/062basic_plotting.ipynb.py:115]: Yes, I was confused about the location. It does live in the Python section in the doctoral course, so maybe it should live in the Python section here too.

**Comment by Saransh-cpp** [comment]
Thanks for the review, @dpshelio! All the suggestions should be in now 🚀

**Review by dpshelio — APPROVED** [review]
:raised_hands:

**Pull request merged**